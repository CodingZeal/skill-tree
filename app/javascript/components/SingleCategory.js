import React from "react";

class SingleCategory extends React.Component {
  render() {
    const { index, rating } = this.props;
    return (
      <div className="singleCategory">
        <h2 id="score">{rating.score}</h2>
        <h2 id="category_name"> {rating.category.category_name}</h2>
      </div>
    );
  }
}

export default SingleCategory;
