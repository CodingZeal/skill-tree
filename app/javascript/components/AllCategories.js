import React from "react";
import SingleCategory from "./SingleCategory";

class AllCategories extends React.Component {
  render() {
    const { myRatings } = this.props;
    const categoryList = myRatings.map((rating, index) => {
      return <SingleCategory rating={rating} index={index} />;
    });
    return <div className="allCategories">{categoryList}</div>;
  }
}

export default AllCategories;
