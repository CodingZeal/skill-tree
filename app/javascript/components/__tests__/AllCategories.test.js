import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import AllCategories from "../AllCategories";
import SingleCategory from "../SingleCategory";

Enzyme.configure({ adapter: new Adapter() });

describe("<AllCategorties>", () => {
  let myRatings = [0, 0, 0];
  const wrapper = shallow(<AllCategories myRatings={myRatings} />);
  it("renders without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("categoryList will match the length of myRatings", () => {
    expect(wrapper.find(SingleCategory)).toHaveLength(myRatings.length);
  });
});
