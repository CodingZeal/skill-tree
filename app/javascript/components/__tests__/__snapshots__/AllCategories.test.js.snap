// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AllCategorties> renders without crashing 1`] = `
<div
  className="allCategories"
>
  <SingleCategory
    index={0}
    rating={0}
  />
  <SingleCategory
    index={1}
    rating={0}
  />
  <SingleCategory
    index={2}
    rating={0}
  />
</div>
`;
